plugins/pulp_docker/plugins/importers/upload.py

```diff
@@ -218,12 +218,23 @@ def process_main(self, item=None):
         manifest_type_id = models.Manifest._content_type_id.default
         repo_manifest_ids = repository.get_associated_unit_ids(repo_id, manifest_type_id)
 
-        # check if there is manifest with such id within the queried rpeo
+        # check if there is manifest with such id within the queried repo
+        # since we don't know if the provided digest is of an image manifest or manifes list
+        # we first need to try both
+        #TODO
+        # BTW i do not really like this approach so suggestions are welcome
+        # but i would prefert to do more jon on the server side then ask form the user
+        # details which does not make sense to provide, like if he is providing image manifest
+        # or manifest list becaues he already provide the digest whichi s unique identifier.
         manifest = models.Manifest.objects.filter(digest=digest, id__in=repo_manifest_ids)
         if manifest.count() == 0:
-            raise PulpCodedValidationException(error_code=error_codes.DKR1010,
-                                               digest=digest,
-                                               repo_id=repo_id)
+            manifest_list_type_id = models.ManifestList._content_type_id.default
+            repo_manifest_list__ids = repository.get_associated_unit_ids(repo_id, manifest_list_type_id)
+            manifest = models.ManifestList.objects.filter(digest=digest, id__in=repo_manifest_list__ids)
```

Comment:
Here also I think is an example where it would be more clear to be strict about naming with manifests vs. manifest lists. They are very different objects.